homeassistant/components/hue/switch.py

```diff
@@ -0,0 +1,95 @@
+"""Support for switch platform for Hue resources (V2 only)."""
+from __future__ import annotations
+
+from typing import Any
+
+from aiohue.v2 import HueBridgeV2
+from aiohue.v2.controllers.events import EventType
+from aiohue.v2.controllers.sensors import LightLevelController, MotionController
+from aiohue.v2.models.resource import SensingService
+
+from homeassistant.components.switch import DEVICE_CLASS_SWITCH, SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ENTITY_CATEGORY_CONFIG
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .bridge import HueBridge
+from .const import DOMAIN
+from .v2.entity import HueBaseEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Hue switch platform from Hue resources."""
+    bridge: HueBridge = hass.data[DOMAIN][config_entry.entry_id]
+    api: HueBridgeV2 = bridge.api
+
+    if bridge.api_version == 1:
+        # should not happen, but just in case
+        raise NotImplementedError("Switch support is only available for V2 bridges")
+
+    for controller in [api.sensors.motion, api.sensors.light_level]:
+        # add entities for all scenes
+        @callback
+        def async_add_entity(event_type: EventType, resource: SensingService) -> None:
+            """Add entity from Hue resource."""
+            async_add_entities(
+                [HueSensingServiceEnabledEntity(bridge, controller, resource)]
+            )
+
+        # add all current items in controller
+        for item in controller:
+            async_add_entity(EventType.RESOURCE_ADDED, item)
+
+        # register listener for new items only
+        config_entry.async_on_unload(
+            controller.subscribe(
+                async_add_entity, event_filter=EventType.RESOURCE_ADDED
+            )
+        )
+
+
+class HueSensingServiceEnabledEntity(HueBaseEntity, SwitchEntity):
+    """Representation of a Switch entity from HUE SensingService."""
+
+    _attr_entity_category = ENTITY_CATEGORY_CONFIG
+    _attr_device_class = DEVICE_CLASS_SWITCH
+
+    def __init__(
+        self,
+        bridge: HueBridge,
+        controller: LightLevelController | MotionController,
+        resource: SensingService,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(bridge, controller, resource)
+        self.resource = resource
+        self.controller = controller
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if the switch is on."""
+        return self.resource.enabled
+
+    @property
+    def icon(self) -> str | None:
```

Comment:
We should do something for this in the frontend. Let's not do it in integrations instead.

Reply:
Yeah we should because the default is a lightning flash that looks like something that needs power while this is a configuration option. I tried playing with different device classes but couldn't get rid of the weird icon.

Reply:
removed the icon, the frontend thingie needs fixing for other integrations too